cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/BusinessMetadataStore.java

```diff
@@ -48,11 +56,17 @@
 
   private static final Id.DatasetInstance BUSINESS_METADATA_INSTANCE_ID =
     Id.DatasetInstance.from(Id.Namespace.SYSTEM, "business.metadata");
+  private static final Map<String, String> EMPTY_PROPERTIES = ImmutableMap.of();
+  private static final Set<String> EMPTY_TAGS = ImmutableSet.of();
+
   private final Transactional<BusinessMdsIterable, BusinessMetadataDataset> txnl;
+  private final boolean publishEnabled;
+  private final MetadataChangePublisher changePublisher;
```

Comment:
yes, my bad. This logic got a bit jumbled up because I initially had the publish method inside the transaction, but then pulled it out. This is probably also the reason why I can't get a test to verify disabling of publishing to pass. Should call it a day when the body demands it :).

Will fix and also add a test.